python_modules/libraries/dagster-dg/dagster_dg/context.py

```diff
@@ -450,16 +447,20 @@ def environment_desc(self) -> str:
     def pyproject_toml_path(self) -> Path:
         return self.root_path / "pyproject.toml"
 
-    def get_path_for_module(self, module_name: str) -> Path:
-        with ensure_loadable_path(self.root_path):
-            parts = module_name.split(".")
-            for i in range(len(parts)):
-                leading_module_name = ".".join(parts[: i + 1])
-                if not is_package_installed(leading_module_name):
-                    raise DgError(
-                        f"Module `{leading_module_name}` is not installed in the current environment."
-                    )
-            return Path(get_path_for_module(module_name))
+    def get_path_for_local_module(self, module_name: str) -> Path:
+        if not self.is_project and not self.is_component_library:
+            raise DgError(
+                "`get_path_for_local_module` is only available in a project or component library context"
+            )
+        if not module_name.startswith(self.root_module_name):
+            raise DgError(f"Module `{module_name}` is not part of the current project.")
+        relative_path = Path(*module_name.split("."))
```

Comment:
seems like you should create an intermediate variable that is `self.root_path / relative_path`